src/middleware/errorHandler.js

```diff
@@ -133,9 +133,8 @@ const filterSecrets = (error, req, res, next) => {
 		req.originalUrl = filterQuery(req.originalUrl);
 		req.body = filter(req.body);
 		error.data = filter(error.data);
-		if (error.catchedError && error.catchedError.options && error.catchedError.options.form) {
-			error.catchedError.options.form = filter(error.catchedError.options.form);
-		}
+		error.catchedError = filter(error.catchedError);
+		error.options = filter(error.options);
```

Comment:
@robertstrz Why where `error= filter(error); ` reverte?

Reply:
Because it changes the error object type and then in node_modules/@feathers/express/lib/error-handler.js it crashed at line 85 . 
```
 let output = Object.assign({}, error.toJSON());

Reply:
okay, that is not good it dosn't solve the problem then.

Reply:
And maybe it create the object members error.catch Error and error.options or trigger for this an not catched exception, if the where not there.

Reply:
As far as I can see catchedError is added in src/app.hooks.js line 120
So, what should be done here ?

Reply:
What i mean is to prevent an exception loop. 
Like: 
Exception (catched by global handler) go to log function.
Log fuction call the filterSecrets function.
Function filterSecrets  throws an exception, because error.options dosn't exist.
Exception (catched by global handler) go to log function.
Log fuction call the filterSecrets function.
Function filterSecrets  throws an exception, because error.options dosn't exist.
.... never ending story ....

Reply:
The function catchedError  ( in src/app.hooks.js line 120 ) should be the last result and fallback.
Errors and Exceptions should and must be handled by the developer in his implementation.
At the best case near to where the where thrown.

Reply:
Yes agree, but as we discussed it here https://ticketsystem.schul-cloud.org/browse/SC-4553 the scope of this task was to filter it from the server logs and then fix it in separate task, when the concept for exception handling will be ready.

Reply:
The change request we discuss know are that the two lines you add should not create an error/exception loop.

Reply:
okay, after a short explanation by @robertstrz, how this is handled by node.js. The result is approved.